### ISSUE TITLE
Terraform plan gets stuck at "Resource module.kms.google_kms_crypto_key.key[0] has lifecycle.prevent_destroy set, but the plan calls for this resource to be destroyed."

### ISSUE THREAD

**Issue by jumanabohra** [terraform-google-modules/terraform-google-kms#95]
### TL;DR

I have manually destroyed a KMS key version in GCP. When I try to create plan for kms using terraform. It keeps giving the error that lifecycle.prevent_destroy is set. I am not trying to destroy anything anymore. I am only trying to create a new KMS key version and set encrypters and decryters for it.

### Expected behavior

The existing key creates a new version with my given decrypters and encrypters.

### Observed behavior

Terraform plan fails stating it cannot destroy the instance. It should ideally not destroy anything

### Terraform Configuration

```hcl
module "kms" {
  source  = "terraform-google-modules/kms/google"
  version = "~> 2.2"

  project_id          = local.project
  location            = "global"
  keyring             = "containersign-keyring"
  keys                = ["containersign-key"]
  key_algorithm       = "RSA_SIGN_PSS_4096_SHA512"
  key_rotation_period = null
  decrypters = [ ... ]
  encrypters = [ ... ]
  set_decrypters_for = [ "containersign-key" ]
  set_encrypters_for = ["containersign-key" ]
}
```

### Terraform Version

```sh
Terraform v1.5.2
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.47.0
```

### Additional information

The output of  $ terraform plan -lock=false

```
data.google_project.project: Reading...
module.kms.google_kms_key_ring.key_ring: Refreshing state... [id=projects/fzo-uapl/locations/global/keyRings/containersign-secret-keyring]     
module.kms.google_kms_crypto_key.key[0]: Refreshing state... [id=projects/fzo-uapl/locations/global/keyRings/containersign-secret-keyring/cryptoKeys/containersign-secret-key]
data.google_project.project: Read complete after 1s [id=projects/fzo-uapl]

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Instance cannot be destroyed
│
│   on .terraform/modules/kms/main.tf line 27:
│   27: resource "google_kms_crypto_key" "key" {
│
│ Resource module.kms.google_kms_crypto_key.key[0] has lifecycle.prevent_destroy set, but the plan calls for this resource to be destroyed. To 
│ avoid this error and continue with the plan, either disable lifecycle.prevent_destroy or reduce the scope of the plan using the -target      
│ flag.
```
